types/hapi/index.d.ts

```diff
@@ -571,10 +571,10 @@ export interface ServerEventCriteria {
  * Related to [See docs](https://hapijs.com/api/16.1.1#servermethodname-method-options)
  */
 export interface ServerMethod {
-    /** Not possible to improve this typing due to this unresolvable issue:  https://github.com/Microsoft/TypeScript/issues/15190 */
-    (...args: (any | ServerMethodNext)[]): void;
     /** the method must return a value (result, Error, or a promise) or throw an Error. */
     (...args: any[]): any | Error | Promise<any>;
+    /** Not possible to improve this typing due to this unresolvable issue:  https://github.com/Microsoft/TypeScript/issues/15190 */
+    (...args: (any | ServerMethodNext)[]): void;
```

Comment:
ok i see the commit. in the future consider rebasing when merging master into your fork.